### PR TITLE
Fix BinaryFormatter AoT test differences

### PR DESCRIPTION
**Pull request by morganbr**
Fix serialization tests to handle a couple of AoT differences:
1. AoT doesn't return ByteEqualityComparer from EqualityComparer<byte>.Default, so create it with reflection instead
2. CoreCLR changes the _version field of List differently than both AoT and NetFX, so pass arrays to the IEnumerable constructor of a few collections, which causes the _version fields to agree. Also updates the stored blobs to reflect this.

### CODE REVIEW

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs [hunk 1]
```diff
@@ -262,7 +275,7 @@ public static IEnumerable<object[]> SerializableObjects_MemberData()
...
...AAAEwAAABNAAAATgAAAE8AAABQAAAAUQAAAFIAAABTAAAAVAAAAFUAAABWAAAAVwAAAFgAAABZAAAAWgAAAFsAAABcAAAAXQAAAF4AAABfAAAAYAAAAGEAAABiAAAAYwAAAGQAAABlAAAAZgAAAGcAAABoAAAAaQAAAGoAAABrAAAAbAAAAG0AAABuAAAAbwAAAHAAAABxAAAAcgAAAHMAAAB0AAAAdQAAAHYAAAB3AAAAeAAAAHkAAAB6AAAACw==", "AAEAAAD/////AQAAAAAAAAAEAQAAAH5TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5MaXN0YDFbW1N5c3RlbS5JbnQzMiwgbXNjb3JsaWIsIFZlcnNpb249NC4wLjAuMCwgQ3VsdHVyZT1uZXV0cmFsLCBQdWJsaWNLZXlUb2tlbj1iNzdhNWM1NjE5MzRlMDg5XV0DAAAABl9pdGVtcwVfc2l6ZQhfdmVyc2lvbgcAAAgICAkCAAAAewAAAAAAAAAPAgAAAHsAAAAIAAAAAAEAAAACAAAAAwAAAAQAAAAFAAAABgAAAAcAAAAIAAAACQAAAAoAAAALAAAADAAAAA0AAAAOAAAADwAAABAAAAARAAAAEgAAABMAAAAUAAAAFQAAABYAAAAXAAAAGAAAABkAAAAaAAAAGwAAABwAAAAdAAAAHgAAAB8AAAAgAAAAIQAAACIAAAAjAAAAJAAAACUAAAAmAAAAJwAAACgAAAApAAAAKgAAACsAAAAsAAAALQAAAC4AAAAvAAAAMAAAADEAAAAyAAAAMwAAADQAAAA1AAAANgAAADcAAAA4AAAAOQAAADoAAAA7AAAAPAAAAD0AAAA+AAAAPwAAAEAAAABBAAAAQgAAAEMAAABEAAAARQAAAEYAAABHAAAASAAAAEkAAABKAAAASwAAAEwAAABNAAAATgAAAE8AAABQAAAAUQAAAFIAAABTAAAAVAAAAFUAAABWAAAAVwAAAFgAAABZAAAAWgAAAFsAAABcAAAAXQAAAF4AAABfAAAAYAAAAGEAAABiAAAAYwAAAGQAAABlAAAAZgAAAGcAAABoAAAAaQAAAGoAAABrAAAAbAAAAG0AAABuAAAAbwAAAHAAAABxAAAAcgAAAHMAAAB0AAAAdQAAAHYAAAB3AAAAeAAAAHkAAAB6AAAACw==" } };
```
Comment (ViktorHofer) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: Could you please add a comment which describes the issue when using the IEnumerable? Thanks a lot!

**Comment by morganbr** [comment]
CC @dotnet/corert-contrib

**Comment by jkotas** [comment]
> CoreCLR changes the _version field of List differently than both AoT and NetFX

Is this expected? It sounds like a sign of missing bug fixes in uapaot version of List.

**Comment by morganbr** [comment]
We discussed offline that it's an internal implementation difference, not a bug. While converging List may be a good idea, it's not required to clean up a BinaryFormatter test.

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs [hunk 2]
```diff
@@ -95,7 +95,20 @@ public static IEnumerable<object[]> SerializableObjects_MemberData()
             yield return new object[] { EqualityComparer<double?>.Default, new string[] { "AAEAAAD/////AQAAAAAAAAAEAQAAAJMBU3lzdGVtLkNvbGxlY3Rpb25zLkdlbmVyaWMuTnVsbGFibGVFcXVhbGl0eUNvbXBhcmVyYDFbW1N5c3RlbS5Eb3VibGUsIG1zY29ybGliLCBWZXJzaW9uPTQuMC4wLjAsIEN1bHR1cmU9bmV1dHJhbCwgUHVibGljS2V5VG9rZW49Yjc3YTVjNTYxOTM0ZTA4OV1dAAAAAAs=", "AAEAAAD/////AQAAAAAAAAAEAQAAAJMBU3lzdGVtLkNvbGxlY3Rpb25zLkdlbmVyaWMuTnVsbGFibGVFcXVhbGl0eUNvbXBhcmVyYDFbW1N5c3RlbS5Eb3VibGUsIG1zY29ybGliLCBWZXJzaW9uPTQuMC4wLjAsIEN1bHR1cmU9bmV1dHJhbCwgUHVibGljS2V5VG9rZW49Yjc3YTVjNTYxOTM0ZTA4OV1dAAAAAAs=" } };
 
             // Equality comparers which can roundtrip
-            yield return new object[] { EqualityComparer<byte>.Default, new string[] { "AAEAAAD/////AQAAAAAAAAAEAQAAAC9TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5CeXRlRXF1YWxpdHlDb21wYXJlcgAAAAAL", "AAEAAAD/////AQAAAAAAAAAEAQAAAC9TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5CeXRlRXF1YWxpdHlDb21wYXJlcgAAAAAL" } };
+            object byteEqualityComparer;
+
+            // .NET Native doesn't create ByteEqualityComparer with EqualityComparer.Default, so create it through reflection
+            if(PlatformDetection.IsNetNative)
```
Comment (krwq) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: nit: `if<space>`

Comment (morganbr) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: Fixed.

**Comment by morganbr** [comment]
@krwq , there might be an argument for adding a third .NET Native blob for each of these, but mismatches against CoreCLR have caught some more interesting bugs, just not these ones.

**Pull request merged**